consumption.tex

```diff
@@ -0,0 +1,89 @@
+
+We adapt a modified version of this scheme for our purposes, to achieve our own 1-way peg by making use of Proof-of-Burn. Since we may be dealing with a source blockchain which doesn't have smart contract functionality, we cannot rely on events being smart contract events. Thus, we define our event as a simple value transfer described by (a) a transaction id \textsf{txid}, (b) a receiving address \textsf{receiver} and (c) an amount \textsf{amount}. Thus our event takes the form of the tuple $e = (\textsf{txid}, \textsf{receivingAddress}, \textsf{amount})$. We remark that simple value transfers are supported by all popular cryptocurrencies, allowing a verifier to process burns from a wide range of source blockchains. We also remark that for now this event type does not yet distinguish between burn and non-burn addresses. Changing the type of event a verifier verifies of course means that the verification function $\textsf{verify}$ also needs to be defined appropriately.
+
+\import{./}{algorithms/alg.verify-tx.tex}
+
+Before looking at how the verification works for the whole proof $\beta$ we focus on the part of the proof pertaining to the transaction, specifically $\alpha = \sf(tx, txInclusion, blk)$. Verifying this part of the proof for some event $e$ is straightforward by making use of the blockchain transaction protocol in Definition~\ref{def:blockchain-tx}. We ensure that the transaction \textsf{tx} indeed contains the value transfer claimed in the provided event $e$, and that the transaction is really contained in the block provided. The function, \textsf{verify-tx}, can be seen in detail in Algorithm~\ref{alg.verify-tx}.
+
+\begin{definition}[Merkle Tree protocol]
+  A \emph{merkle tree} protocol $\mathcal{MT}$ consists of two functions $\mathcal{MT}_t(x_1, \dots, x_t)$ and $\VerMT_t(root, x, \pi)$ which work as follows:
+
+  \begin{itemize}
+    \item $\mathcal{MT}_t(x_1, \dots, x_t)$: Given elements $x_1, \dots, x_t$, return the root of the Merkle Tree with exactly $t$ leaves, where for each $1 \le i \le t$, the $i$-th leaf corresponds to $x_i$.
+
+    \item $\VerMT(root, x, \pi)$: Given a Merkle Tree root, an element $x$ and an inclusion proof $\pi$, return $\sf{true}$ if and only if $\pi$ is a valid proof for the fact that $x$ is contained in the tree rooted at $root$.
+  \end{itemize}
+\end{definition}
```

Comment:
I feel this definition can be removed. Just say that "the verification function ensures that the transaction id belongs to the Merkle Tree whose root is $\mathcal{M}$". The functions \mathcal{T}_t and \VerMT can also just be referred to in the text by saying "A merkle tree root is constructed from a tree with block ids as leaves".